### PR TITLE
[dadata] initial version

### PR DESCRIPTION
**Pull request by kTelyakov**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,7 +1,7 @@
 root = true
 
 [*]
-indent_size = 4
+indent_size = 2
```
Comment (sandersn) [inline .editorconfig:4]: this should stay at 4

File: types/dadata/tslint.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "@definitelytyped/dtslint/dt.json",
+  "rules": {
+    "comment-format": false,
+    "export-just-namespace": false,
+    "interface-name": false,
+    "no-any-union": false,
+    "no-unnecessary-class": false
+  }
```
Comment (sandersn) [inline types/dadata/tslint.json:9]: Are these needed? New packages should leave all the default lint rules on unless there's a good reason. I don't see any classes in the code, so I guess that no-unnecessary-class, at least, is not needed at all.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (kTelyakov) [reply .editorconfig]: thank you. Fixed

File: types/dadata/tslint.json (hunk 2 above)
Comment (kTelyakov) [reply types/dadata/tslint.json]: I removed the extra rules that I laid down for the future. Thank you

File: types/dadata/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        local: Nullable<string>;
+        domain: Nullable<string>;
+        type: null;
+        qc: null;
+        source: null;
+    }
+}
+
+export = DaData;
+export as namespace DaData;
```
Comment (peterblazejewicz) [inline types/dadata/index.d.ts:253]:
```suggestion
```
IMO it's just old school (check the no bracket syntax btw) CJS module:
https://unpkg.com/browse/dadata@0.0.3/index.js

File: types/dadata/tslint.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "@definitelytyped/dtslint/dt.json",
+  "rules": {
+    "export-just-namespace": false
+  }
```
Comment (peterblazejewicz) [inline types/dadata/tslint.json:5]:
```suggestion
  "extends": "@definitelytyped/dtslint/dt.json"
```
add this rule inline, if still required

Comment (kTelyakov) [reply types/dadata/tslint.json:5]: @peterblazejewicz i'm sorry, but build cant get completed, if i remove "export-just-namespace": false

File: types/dadata/index.d.ts (hunk 3 above)
Comment (kTelyakov) [reply types/dadata/index.d.ts:253]: ok, fixed

Comment (kTelyakov) [reply types/dadata/index.d.ts]: @peterblazejewicz i added your suggested change to index.d.ts, then build was broke ((

File: types/dadata/tslint.json (hunk 4 above)
Comment (peterblazejewicz) [reply types/dadata/tslint.json:5]: hi,

so here is a code from readme (tests.ts here)
```ts
import DaData = require('dadata');

const dadata_account_1 = new DaData('<API_KEY_1>', '<SECRET_KEY_1>');
const dadata_account_2 = new DaData('<API_KEY_2>', '<SECRET_KEY_2>');
dadata_account_1('address', ['мск сухонска 11/-89'], (err, res) => {});
```
and this is sample implementation missing your current code. It seems you've just added interfaces for models, nothing for actually using export symbols (which are a function with constructor function), something like:

```ts
...

        source: null;
    }

    interface Results {
        // TBD
    }

    interface Service {
        (
            apiType: string,
            queries: string[],
            callback: (err: Error | undefined, res: Results | undefined) => void,
        ): void;
    }
}

interface DaDataConstructor {
    new (apiKey: string, secretKey: string): DaData.Service;
    (apiKey: string, secretKey: string): DaData.Service;
}

declare const DaData: DaDataConstructor;

export = DaData;
```

Comment (kTelyakov) [reply types/dadata/tslint.json]: @peterblazejewicz its my fault - i attempted to add definitions for api from service dadata (https://dadata.ru/api/), i dont saw package "dadata" exist in npm )) Ive been already renamed my module - can you check it ?

Comment (kTelyakov) [reply types/dadata/tslint.json]: @peterblazejewicz my definitions only for responses from https://dadata.ru/api/, not for any existing package in npm )

Comment (peterblazejewicz) [reply types/dadata/tslint.json]: ops, not sure why dadata got into the discussion, please disregard my commens and sample as not relevant

**Comment by kTelyakov** [comment]
Hello ! Anybody can see my PR ? )

File: types/dadata-api/tslint.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "@definitelytyped/dtslint/dt.json",
+  "rules": {
+    "export-just-namespace": false
+  }
```
Comment (sheetalkamat) [inline types/dadata-api/tslint.json:5]:
```suggestion
```
Please fix the errors instead of these lines

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Pull request closed without merging**